azext_iot/central/models/template.py

```diff
@@ -15,30 +15,89 @@ def __init__(self, template: dict):
+                self.component_schema_names = self._extract_schema_names(
+                    self.components
+                )
+
         except:
             raise CLIError("Could not parse iot central device template.")
 
-    def get_schema(self, name, interface_name=""):
-        # interface_name has been specified, do a pointed lookup
-        if interface_name:
-            interface = self.interfaces.get(interface_name, {})
-            return interface.get(name)
+    def get_schema(self, name, is_component=False, identifier="") -> dict:
+        if is_component:
+            if identifier:
+                # identifier specified, do a pointed lookup
+                component = self.components.get(identifier, {})
+                return component.get(name)
+
+            # find first matching name in any component
+            for component in self.components.values():
+                schema = component.get(name)
+                if schema:
+                    return schema
+        else:
+            # identifier specified, do a pointed lookup
+            if identifier:
+                interface = self.interfaces.get(identifier, {})
+                return interface.get(name)
 
-        # find first matching name in any interface
-        for interface in self.interfaces.values():
-            schema = interface.get(name)
-            if schema:
-                return schema
+            # find first matching name in any interface
+            for interface in self.interfaces.values():
+                schema = interface.get(name)
+                if schema:
+                    return schema
 
         # not found
         return None
 
+    def _extract_components(self, template: dict) -> dict:
+        try:
+            dcm = template.get("capabilityModel", {})
+            if dcm.get("contents"):
+                rootContents = dcm.get("contents", {})
+                components = [
+                    entity
+                    for entity in rootContents
+                    if entity.get("@type") == "Component"
+                ]
+
+                if components:
+                    return {
+                        component["name"]: self._extract_schemas(component)
+                        for component in components
+                    }
+
+        except Exception:
+            details = "Unable to extract schema for component from template '{}'.".format(
+                self.id
+            )
+            raise CLIError(details)
+
+    def _extract_root_interface_contents(self, dcm: dict):
+        rootContents = dcm.get("contents", {})
+        contents = [
+            entity for entity in rootContents if entity.get("@type") != "Component"
+        ]
+
+        return {"@id": dcm.get("@id", {}), "schema": {"contents": contents}}
+
     def _extract_interfaces(self, template: dict) -> dict:
         try:
+
+            interfaces = []
             dcm = template.get("capabilityModel", {})
```

Comment:
capabilityModel is removed in DTDL v2.

Reply:
wont fix - current public api supports both V1 and V2. Templates that are prior to DTDLV2 service side update will continue to remain as V1 and  newer templates created will be V2 format.